cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+
+    std::set<vertex_t> vertices(dup_vertices.begin(), dup_vertices.end());
+
+    auto n_dropped = 1;
+
+    auto edge_first = thrust::make_zip_iterator(thrust::make_tuple(h_src.begin(), h_dst.begin()));
+
+    auto weighted_edge_first =
+      thrust::make_zip_iterator(thrust::make_tuple(h_src.begin(), h_dst.begin(), h_wgt.begin()));
+
+    thrust::sort(thrust::host, edge_first, edge_first + h_src.size());
+
+    while (n_dropped > 0) {
+      n_dropped = 0;
+      std::set<vertex_t> seen;
+      // Go over all the vertices
+      for (auto u = vertices.begin(); u != vertices.end(); ++u) {
+        // Find all neighbours of u
+        vertex_t idx_start   = 0;
+        vertex_t idx_end     = 0;
+        auto itr_lower_range = thrust::lower_bound(thrust::host, h_src.begin(), h_src.end(), *u);
+
+        idx_start = thrust::distance(h_src.begin(), itr_lower_range);
+
+        if (*itr_lower_range == *u) {
+          auto itr_upper_range =
+            thrust::upper_bound(thrust::host, itr_lower_range, h_src.end(), *u);
+
+          idx_end = thrust::distance(itr_lower_range, itr_upper_range);
+        }
+
+        std::set<vertex_t> nbrs_u;
+
+        for (edge_t i = idx_start; i < idx_start + idx_end; ++i) {
+          nbrs_u.insert(*(h_dst.begin() + i));
+        }
+
+        seen.insert(*u);
+
+        std::set<vertex_t> new_nbrs;
+
+        std::set_difference(nbrs_u.begin(),
+                            nbrs_u.end(),
+                            seen.begin(),
+                            seen.end(),
+                            std::inserter(new_nbrs, new_nbrs.end()));
+
+        // Finding the neighbors of v
+        for (auto v = new_nbrs.begin(); v != new_nbrs.end(); ++v) {
+          auto itr_lower_range = thrust::lower_bound(thrust::host, h_src.begin(), h_src.end(), *v);
+
+          idx_start = thrust::distance(h_src.begin(), itr_lower_range);
+
+          if (*itr_lower_range == *v) {
+            auto itr_upper_range =
+              thrust::upper_bound(thrust::host, itr_lower_range, h_src.end(), *v);
+
+            idx_end = thrust::distance(itr_lower_range, itr_upper_range);
+          }
+          std::set<vertex_t> nbrs_v;
+
+          for (edge_t i = idx_start; i < idx_start + idx_end; ++i) {
+            nbrs_v.insert(*(h_dst.begin() + i));
+          }
+
+          std::set<vertex_t> nbr_intersection_u_v;
+          // Find the intersection of nbr_u and nbr_v
+          std::set_intersection(nbrs_u.begin(),
+                                nbrs_u.end(),
+                                nbrs_v.begin(),
+                                nbrs_v.end(),
+                                std::inserter(nbr_intersection_u_v, nbr_intersection_u_v.end()));
+
+          if (nbr_intersection_u_v.size() < (k - 2)) {
+            auto edge = thrust::make_tuple(vertex_t{*u}, vertex_t{*v});
+
+            if (edge_weight_view) {
+              auto weighted_edge_last = thrust::remove_if(
```

Comment:
std::remove_if

Reply:
This part of the code is outdated and the reference implementation is only using std vectors